diff --git a/latex-svg-hakyll.cabal b/latex-svg-hakyll.cabal
index c3cf24b..f4aa18e 100644
--- a/latex-svg-hakyll.cabal
+++ b/latex-svg-hakyll.cabal
@@ -30,8 +30,8 @@ library
   exposed-modules:  Hakyll.Contrib.LaTeX
   build-depends:
     , base              >=4.10      && <4.14
-    , hakyll            ^>=4.12.5.2 || ^>=4.13.0.0
+    , hakyll            ^>=4.12.5.2 || ^>=4.13.0.0 || ^>=4.14.0.0
     , latex-svg-image   ^>=0.2
     , latex-svg-pandoc  ^>=0.2
     , lrucache          ^>=1.2
-    , pandoc-types      ^>=1.17.6.1 || ^>=1.20
+    , pandoc-types      ^>=1.17.6.1 || ^>=1.20 || ^>= 1.21 || ^>= 1.22
